contrib/mesos/pkg/minion/tasks/task.go

```diff
@@ -240,7 +247,8 @@ type exitError interface {
 	Sys() interface{}
 }
 
-func taskRunning(t *Task) taskStateFn {
+func taskRunning(t *Task) (nextState taskStateFn) {
```

Comment:
This seemed cleaner, more DRY. Otherwise basically the same code blob gets copy-pasted elsewhere. I'm happy to refactor the loop into a separate func that's more easily testable.